### ISSUE TITLE
Vertical orientation - responsive commits column

### ISSUE THREAD

**Issue by Otiel** [ForkIssues/TrackerWin#193]
_As suggested by @jerone in [#55](https://github.com/ForkIssues/TrackerWin/issues/55#issuecomment-468231395):_

When using the alternative vertical layout (or three-columns view), this would be nice if the commits column was single-line when there is enough space. See this screenshot for instance:

![screenshot](https://user-images.githubusercontent.com/55841/53261068-63179500-36d3-11e9-9cd6-0792cf6cd7b9.png)

The suggestion is to make this column responsive and display commits info on two lines only when there's not enough space.

**Comment by DanPristupov** [comment]
I think it's not possible to make a responsive grid in WPF. We can only introduce a 'one-line' option

**Comment by Namek** [comment]
If not responsively then with a simple manual switch. Table view is much easier to look through for me, usually. I got ultra wide screen and want to take advantage of it.

**Comment by Otiel** [comment]
>I think it's not possible to make a responsive grid in WPF.

How do you mean "a responsive grid"? WPF offers plenty ways of doing "responsive design".

**Comment by DanPristupov** [comment]
@Otiel is it possible to change row height of a ListView dynamically, relying on the available width?

Update: in the virtual mode

**Comment by Otiel** [comment]
@DanPristupov "row of a ListBox" = `ListBoxItem` ?

**Comment by DanPristupov** [comment]
Sorry, I updated my post. Not ListBox, but ListView. So, yes, ListViewItem

**Comment by Otiel** [comment]
Well, quick hack shows it is possible:

_Window.xaml_:
```
<ListView Name="listView1" SizeChanged="ListView_SizeChanged">
    <ListViewItem Background="#FFE9F5FF">
            <TextBlock>Commit 3</TextBlock>
    </ListViewItem>
    <ListViewItem Background="#FFECD9D9">
            <TextBlock>Commit 2</TextBlock>
    </ListViewItem>
    <ListViewItem Background="#FFE9F5FF">
            <TextBlock>Commit 1</TextBlock>
    </ListViewItem>
</ListView>
```

_Window.xaml.cs_:
```
private void ListView_SizeChanged(object sender, SizeChangedEventArgs e) {
    int height = listView1.ActualWidth > 300 ? 80 : 50;
    foreach (ListViewItem item in listView1.Items) {
        item.Height = height;
    }
}
```


![gif](https://user-images.githubusercontent.com/3851902/54133113-c7e22780-4415-11e9-97bf-5e14ac9a884b.gif)

There's probably a better way to do this in xaml only.

**Comment by DanPristupov** [comment]
Can you try with 15000 rows please?

**Comment by Otiel** [comment]
Seems poorly optimized, but I have more or less the same results without the height change.

# With height change

_Window.xaml_:
```
<ListView Name="listView1" SizeChanged="ListView_SizeChanged" />
```

_Window.xaml.cs_:
```
public partial class MainWindow: Window {
    Double _listViewPreviousWidth = 0;

    public MainWindow() {
        InitializeComponent();
        var brushBlue = new SolidColorBrush(Color.FromRgb(233, 245, 255));
        var brushRed = new SolidColorBrush(Color.FromRgb(236, 217, 217));

        for (int i = 0; i < 15000; i++) {
            ListViewItem item = new ListViewItem {
                Background = i % 2 == 0 ? brushBlue : brushRed,
                Content = "Commit " + i,
            };
            listView1.Items.Add(item);
        }
    }

    private void ListView_SizeChanged(object sender, SizeChangedEventArgs e) {
        if (listView1.ActualWidth < 300 && _listViewPreviousWidth >= 300 || 
            listView1.ActualWidth >= 300 && _listViewPreviousWidth < 300) {
            int height = listView1.ActualWidth < 300 ? 40 : 20;
            foreach (ListViewItem item in listView1.Items) {
                item.Height = height;
            }
        }
        _listViewPreviousWidth = listView1.ActualWidth;
    }
}
```

![gif with height change](https://user-images.githubusercontent.com/3851902/54135986-28279800-441b-11e9-8dde-f9d9eabd6ea1.gif)

# Without height change

_Window.xaml_:
```
<ListView Name="listView1" />
```

![gif without height change](https://user-images.githubusercontent.com/3851902/54136065-56a57300-441b-11e9-8e74-d1aabe5fcfa0.gif)

**Comment by whins** [comment]
Hi
How about this layout:
![layout](https://user-images.githubusercontent.com/2223328/54593401-253e3000-4a37-11e9-8178-789d63947ff9.png)

**Comment by DanPristupov** [comment]
You need to explain the idea at least.

The original vertical layout was made made for people with wide monitors.

I don't see any benefit in this concept though:
- Same number of commits as in usual layout, but less info.
- Less space for commit description.
- More horizontal space for code, which nobody really needs.

**Comment by jerone** [comment]
So, if making the list responsive (switching between one or two lines) is to difficult, what about making it an option. For both the vertical and horizontal commits view.

**Current (two lines):**
![image](https://user-images.githubusercontent.com/55841/54611765-dce33a00-4a57-11e9-983f-dfb6c3cb4e0a.png)

**After clicking the one-line option:**
![image](https://user-images.githubusercontent.com/55841/54611744-d5bc2c00-4a57-11e9-967e-61d985f3c64f.png)

**Comment by DanPristupov** [comment]
@jerone yes, I think we will end up with something like this (but without sha, probably).

**Comment by Otiel** [comment]
@DanPristupov Any ETA for a simple option to disable the two-lines view in vertical orientation?

After spending lots of time on Fork, I realize that the two-lines view is not as easily readable as the one-line view, but I couldn't live without the vertical orientation to read committed files now.

**Comment by DanPristupov** [comment]
Alright, we have ETA of the responsive approach. Tanya has already made a proof of concept which works pretty well so far. We will test it during the week and hopefully make it a part of the 1.33 release.

**Comment by DanPristupov** [comment]
1.33 is out!
- New vertical layout
- Support of issue trackers
- Side by side diff

**Comment by Otiel** [comment]
Oh yeah, that's some good stuff!

Thanks ❤️

**Comment by jerone** [comment]
Awesome! 

Is it possible to decrease the minimum right section more? I feel the current restriction can be lowered by at least 200 pixels.

**Comment by Otiel** [comment]
Reopening this to track the change asked by @jerone

**Comment by sm-g** [comment]
there is empty space after switch orientation:
![Fork_2019-07-05_20-04-59](https://user-images.githubusercontent.com/7280273/60736695-7fb8b900-9f60-11e9-9293-2209eb6ec62d.png)

need to resize to fix columns width

**Comment by jerone** [comment]
Seen this too. 

Haven't had the time to make a decent report and reproduce it.

**Comment by DanPristupov** [comment]
The layout might not be refreshed if you switch orientation multiple times, for example.

**Comment by Otiel** [comment]
Closing this as the original request has been implemented and the [bug reported](https://github.com/ForkIssues/TrackerWin/issues/193#issuecomment-508817814) by @sm-g has been fixed in 1.36 (see #277).

@jerone's [request](https://github.com/ForkIssues/TrackerWin/issues/193#issuecomment-493544744) is now tracked in #359.
